### ISSUE TITLE
[BUG]After closing the ServiceBusSenderClient instance the program still keeps running

### ISSUE THREAD

**Issue by Eugeniu-N** [Azure/azure-sdk-for-java#20660]
**Describe the bug**
I build a new instance of ServiceBusSenderClient using the example in this repo. I send a message (which is successfully delivered to the bus), I close the client but the program keeps running, instead of terminating. After 2 minutes the program is still running, I haven't measured the exact time needed for it to terminate.

**To Reproduce**
Run my code snippet

***Code Snippet***
```
package test;

import java.util.UUID;

import com.azure.core.util.BinaryData;
import com.azure.messaging.servicebus.ServiceBusClientBuilder;
import com.azure.messaging.servicebus.ServiceBusMessage;
import com.azure.messaging.servicebus.ServiceBusSenderClient;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

public class ServiceBusTest {
	private static String connectionString = "";
	private static String queueName = "";
	
	public static void main(String[] args) {
		run();
	}
	
	private static void run() {
		long start = System.currentTimeMillis();
		ServiceBusSenderClient client = new ServiceBusClientBuilder()
				.connectionString(connectionString)
				.sender()
				.queueName(queueName)
				.buildClient();
		System.out.printf("Opened in %.2f seconds\n", (System.currentTimeMillis() - start)/1000.0);
		
		ObjectMapper mapper = new ObjectMapper();
		ObjectNode node = mapper.createObjectNode();
		node.put("id", UUID.randomUUID().toString());
		
		ServiceBusMessage message = new ServiceBusMessage(BinaryData.fromString(node.toString()));
		
		start = System.currentTimeMillis();
		client.sendMessage(message);
		System.out.printf("Sent in %.2f seconds\n", (System.currentTimeMillis() - start)/1000.0);
		
		start = System.currentTimeMillis();
		client.close();
		System.out.printf("Closed in %.2f seconds\n", (System.currentTimeMillis() - start)/1000.0);
	}
}
```

**Expected behavior**
I expect the program to terminate shortly after closing the client. Another issue I have is the time needed to complete the operations. For instance it takes 0.7, 1 and 4 seconds respectively to open, send the message and close the client. I have also used the azure-servicebus 3.6.2 library (which has the exact same issue) but those operations take 0.8, 0.3 and 0.1 seconds. The difference in time is huge

**Setup (please complete the following information):**
 - OS: windows 10
 - IDE : eclipse 6-2020
 - Version of the Library used: 7.2.0

**Comment by joshfree** [comment]
@YijunXieMS PTAL

**Comment by hemanttanwar** [comment]
We have a fix for this and will release this in June month release.

**Comment by YijunXieMS** [comment]
Hi @Eugeniu-N We released 7.2.2 today. Could you upgrade and try?

**Comment by Eugeniu-N** [comment]
I ran the exact same snippet with the 7.2.2 version (made sure to also to do maven > update project on the eclipse project) but the problem of the program not terminating still remains.

I use the library in an azure function though and this doesn't appear to cause problems there because I believe that once a function returns a value the threads spawned by that invocation are immediately destroyed and this is a issue only for actual dedicated servers

**Comment by YijunXieMS** [comment]
Hi @Eugeniu-N my bad. I make a mistake in cherry-picking this fix so it was not included in this release. It will be included in the next release (about 06/10).

**Comment by conniey** [comment]
In our current `main` branch, I can see it exiting after the `retryOptions.tryTimeout` elapsed. (We schedule some pending work to complete on the proton-j reactor after the retryOptions.timeoutOption` elapses.

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Opened in 0.55 seconds
Sent in 0.87 seconds
Closed in 0.02 seconds

Process finished with exit code 0
```

**Comment by ki1729** [comment]
This is fixed in v7.4.0. Client closes returns in 0.02s, and App closes after cleanup timeout
